### ISSUE TITLE
Solved

### ISSUE THREAD

**Issue by muses0229** [autonomousvision/LaRa#13]

**Comment by apchenstu** [comment]
I guess this is because of the coordinate system [alignment](https://github.com/autonomousvision/LaRa/blob/main/dataLoader/instant3d.py#L71-L76), the coordinate is transformed to the first view by default. So you may need to either pass the t`ransform_mats` to the evaluator or disable the transformation.

**Comment by yejr0229** [comment]
I greatly appreciate your quick reply! I try to disable the first view transformation like this:
![企业微信截图_17274522742848](https://github.com/user-attachments/assets/0a3f8784-896c-4478-9b44-7547ee11b5c5)
And I get the GS rendering with elev=-30 azim=315 and in left, it's still not right:
![image](https://github.com/user-attachments/assets/fa3b6651-ab36-4088-82f8-bb6e76a91b0e)

Than I try to pass the transform_mats to the evaluator like this:
![企业微信截图_17274529892073](https://github.com/user-attachments/assets/d55e06fa-ec49-47a2-9f07-b24cf4312c77)

And I still get a wrong result when elev=0 and azim=0:
![image](https://github.com/user-attachments/assets/06ce3503-ded6-4b38-83db-b9f3c65ac66b)

**Comment by apchenstu** [comment]
Have you also disabled it in the dataloader? And the L101-104 is a random pick starting point, you may also need to adjust it according to the coordinate you are using. 
BTW, which dataset are you testing?

**Comment by apchenstu** [comment]
BTW, which dataset are you testing?

**Comment by yejr0229** [comment]
I'm little confused about 'disabled it in the dataloader', is it directly set this to None?
![企业微信截图_17274551682396](https://github.com/user-attachments/assets/36b1391c-aa34-4c01-99c2-e4ab52d5909e)

I'm not using any dataset, just trying to make the rendering from GS and pytorch3d align.

**Comment by apchenstu** [comment]
> I'm little confused about 'disabled it in the dataloader', is it directly set this to None?

Ah, I see. I thought you were using instant3d.py loader.

> I'm not using any dataset, just trying to make the rendering from GS and pytorch3d align.

the current code base uses opencv coordinate system, this [issue](https://github.com/facebookresearch/pytorch3d/issues/1415) may be helpful in converting them

**Comment by yejr0229** [comment]
Thank you once again for your help! I read the issue you mention and seems like the fixing the z-axis of the OpenCV coordinate system and rotating the x-axis and y-axis by 180 degrees results in the P3D coordinate system. I inplement it like this:
![企业微信截图_17275029526077](https://github.com/user-attachments/assets/b106603d-0e65-4060-8599-924e6a88ddd5)
And results are below, the third one is after rotate:
![企业微信截图_17275028976293](https://github.com/user-attachments/assets/d1e0dbe1-3bf3-4c24-b7d4-e0d4bdef6a1c)
I guess the problem it's not caused by the coord, because the coord only affect the object towards.
